internal/acceptance/job_test.go

```diff
@@ -293,6 +293,102 @@ func TestAccJobControlRunState(t *testing.T) {
 	})
 }
 
+// An integration test which creates a table trigger job, verifying that a job run was triggered
+// within 5 minutes of the job creation.
+func TestTableTriggerJob(t *testing.T) {
+	getJobTemplate := func(tableBlock string) string {
+		return `
+		data "databricks_current_user" "me" {}
+		data "databricks_spark_version" "latest" {}
+		data "databricks_node_type" "smallest" {
+			local_disk = true
+		}
+
+		resource "databricks_notebook" "this" {
+			path     = "${data.databricks_current_user.me.home}/Terraform-test"
+			language = "PYTHON"
+			content_base64 = base64encode(<<-EOT
+				# created from ${abspath(path.module)}
+				import time
+
+				display(spark.range(10))
+				time.sleep(3600)
+				EOT
+			)
+		}
+
+		resource "databricks_job" "this" {
+			name = "{var.RANDOM}"
+
+			task {
+				task_key = "a"
+
+				new_cluster {
+					num_workers   = 1
+					spark_version = data.databricks_spark_version.latest.id
+					node_type_id  = data.databricks_node_type.smallest.id
+				}
+
+				notebook_task {
+					notebook_path = databricks_notebook.this.path
+				}
+			}
+
+			trigger {
+				pause_status = "UNPAUSED"
+				table_update {
+					` + tableBlock + `
+				}
+			}
+		}`
+	}
+	previousRunIds := make([]int64, 0)
+	checkIfRunHasStarted := func(ctx context.Context, w *databricks.WorkspaceClient, jobID int64) (bool, error) {
+		runs, err := w.Jobs.ListRunsAll(ctx, jobs.ListRunsRequest{JobId: jobID})
+		assert.NoError(t, err)
+		runIdsMap := make(map[int64]bool)
+		for _, id := range previousRunIds {
+			runIdsMap[id] = true
+		}
+
+		for _, run := range runs {
+			if _, ok := runIdsMap[run.RunId]; !ok && run.State.LifeCycleState == "RUNNING" {
+				previousRunIds = append(previousRunIds, run.RunId)
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+	retryFor := func(ctx context.Context, client *common.DatabricksClient, id string, lastErr error, f func(context.Context, *databricks.WorkspaceClient, int64) (bool, error)) error {
+		ctx = context.WithValue(ctx, common.Api, common.API_2_1)
+		w, err := client.WorkspaceClient()
+		assert.NoError(t, err)
+		jobID, err := strconv.ParseInt(id, 10, 64)
+		assert.NoError(t, err)
+		for i := 0; i < 100; i++ {
+			success, err := f(ctx, w, jobID)
+			if err != nil {
+				return err
+			}
+			if success {
+				return nil
+			}
+			time.Sleep(5 * time.Second)
+		}
+		return lastErr
+	}
+	waitForRunToStart := func(ctx context.Context, client *common.DatabricksClient, id string) error {
+		return retryFor(ctx, client, id, errors.New("timed out waiting for job run to start"), checkIfRunHasStarted)
+	}
+	workspaceLevel(t, step{
+		Template: getJobTemplate(`
+			table_names = ["catalog.schema.table1", "catalog.schema.table2"],
```

Comment:
these tables do not exist in our integration environment, so the job will not trigger

Reply:
we can create a table with the resource `databricks_sql_table`, but to trigger an update would require sending a sql statement

it does feel complicated, so for this PR, you can test locally and confirm that it works. We can add better integration tests in a follow-up PR